src/connection.js

```diff
@@ -717,10 +731,33 @@ class Connection extends EventEmitter {
     });
   }
 
+  // Returns false to apply backpressure.
   sendDataToTokenStreamParser(data) {
     return this.tokenStreamParser.addBuffer(data);
   }
 
+  // This is an internal method that is called from Request.pause().
+  // It has to check whether the passed Request object represents the currently
+  // active request, because the application might have called Request.pause()
+  // on an old inactive Request object.
+  pauseRequest(request) {
+    if (this.isRequestActive(request)) {
+      this.tokenStreamParser.pause();
```

Comment:
> While the driver is in the SENT_ATTENTION state and waiting for the attention response from the server, the application calls Connection.cancel(). The current implementation of the cancel() method considers this an error, writes the debug message "Requests can only be canceled in the SentClientRequest state, not the SentAttention state" and returns false.

I think that's a bug and we should fix that. Opened https://github.com/tediousjs/tedious/issues/545. 

My comment is about after the Request is done and the callback is invoked. At this point the application knows the request is done. I'm not able to come up with an application pattern where there could reasonably be more calls on the Request object other than a bug in the code. Being forgiving here I think hurts than helps the developer by hiding bugs. And there is no need for a isCompleted either, as the callback already gives that information.